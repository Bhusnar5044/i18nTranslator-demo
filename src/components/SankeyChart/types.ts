export type Props = {
    // SankeyChart prop type
    click: () => void;
};
