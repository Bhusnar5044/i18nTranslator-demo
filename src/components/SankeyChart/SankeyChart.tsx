import { Box, Text } from '@Lib';
import { useAppSelector } from '@store';
import { SalaryState } from '@store/salary';
import { theme } from '@theme';
import colors from '@theme/colors';
import { FC, memo } from 'react';
import { Chart } from 'react-google-charts';

export const options = {
    sankey: {
        link: { color: { fill: colors.success.light } },
        node: {
            label: { color: theme.colors.info.dark },
        },
    },
};

export const SankeyChart: FC = memo(() => {
    const {
        sankeyData: { links, nodes },
    } = useAppSelector((state) => state.salary) as SalaryState;
    const data = [['from', 'to', 'weight'], ...links.map((item) => Object.values(item))];
    return (
        <Box flexDirection="column">
            <Text textVariant="h3">Sankey Chart</Text>
            <Chart chartType="Sankey" width="100%" height="100vh" data={data} options={options} />
        </Box>
    );
});

SankeyChart.displayName = 'SankeyChart';
