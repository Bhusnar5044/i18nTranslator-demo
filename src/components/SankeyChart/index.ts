export { SankeyChart as default } from './SankeyChart';
