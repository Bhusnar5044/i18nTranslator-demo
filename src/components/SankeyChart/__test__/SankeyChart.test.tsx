import { renderWithStore } from '@test-utils';
import { SankeyChart } from '../SankeyChart';

describe('SankeyChart', () => {
    it('should render properly', () => {
        const { container } = renderWithStore(<SankeyChart />);
        expect(container).toMatchSnapshot();
    });
});
