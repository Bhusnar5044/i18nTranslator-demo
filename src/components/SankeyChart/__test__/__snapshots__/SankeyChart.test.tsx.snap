// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SankeyChart should render properly 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
}

.c1 {
  margin: 0;
  color: inherit;
  font-size: 2.4rem;
  font-weight: 400;
  letter-spacing: -0.04rem;
  line-height: 3.2rem;
  text-align: initial;
}

<div>
  <div
    class="c0"
    flexdirection="column"
  >
    <h3
      class="c1"
      textalign="initial"
      textcolor="inherit"
      textvariant="h3"
    >
      line2
    </h3>
  </div>
</div>
`;
